### PR TITLE
Simplify sampleNoReplace function

### PR DESCRIPTION
**Pull request by urkerab**
Short and sweet!

### CODE REVIEW

**Comment by Slayer95** [comment]
There is no real gain from this change as the method name is self-explanatory.
I specifically avoided that version due to lower performance when optimizing randbat set generators at 1deedc5952e

**Comment by Zarel** [comment]
@Slayer95, you should probably comment that your version is faster than `.splice`

**Comment by urkerab** [comment]
OK, so is there at least a need for it to keep the array in order?

**Comment by ascriptmaster** [comment]
Now that you mention it... no, I don't think so. So we can probably achieve `ϴ(1)` by filling in the entry with the last position instead of our current `O(N)` efficiency?

**Comment by Slayer95** [comment]
Good point, @urkerab, @ascriptmaster . Nice catch!

**Comment by ascriptmaster** [comment]
More accurately: if you aren't using the array for anything other than the sample function, then it doesn't matter.

**Comment by urkerab** [comment]
Ugh, due to a nasty bug in GitHub I think I've gone and locked this PR so it can't be reopened. Sorry about that. I'll have open a new PR later.

File: data/scripts.js [hunk 1]
```diff
@@ -636,14 +636,7 @@ exports.BattleScripts = {
 		return isValid;
 	},
 	sampleNoReplace: function (list) {
-		let length = list.length;
-		let index = this.random(length);
-		let element = list[index];
-		for (let nextIndex = index + 1; nextIndex < length; index += 1, nextIndex += 1) {
```
Comment (urkerab) [inline data/scripts.js]: Next thing you'll be telling me that `index += 1` is faster than `++index`... wait, yes, it's 3.5% faster; now I've seen it all...

**Comment by ascriptmaster** [comment]
@urkerab I don't know how I reopened it but here ya go?

**Comment by urkerab** [comment]
Latest push just removes the loop as discussed. (Also I have no problem with the naming of the function; even if the `splice` code was performant, I would still want it as a single-line function.)

File: data/scripts.js (hunk 1 above)
Comment (urkerab) [inline data/scripts.js]: (`++index` is 5% faster than `index += ` in Node 4.0.0; `index++` is now only 1% slower)

**Comment by Zarel** [comment]
Can you make a comment (the `//` kind of comment) about why it doesn't use splice?

**Comment by Zarel** [comment]
(I asked @Slayer95 to, but it doesn't look like he's gotten around to it yet)

**Comment by urkerab** [comment]
Comment added.

**Pull request merged**